src/controllers/hid/hidcontroller.cpp

```diff
@@ -16,36 +16,7 @@
 #include "controllers/controllerdebug.h"
 #include "util/time.h"
 
-HidReader::HidReader(hid_device* device)
-        : QThread(),
-          m_pHidDevice(device) {
-}
-
-HidReader::~HidReader() {
-}
-
-void HidReader::run() {
-    m_stop = 0;
-    unsigned char *data = new unsigned char[255];
-    while (m_stop.load() == 0) {
-        // Blocked polling: The only problem with this is that we can't close
-        // the device until the block is released, which means the controller
-        // has to send more data
-        //result = hid_read_timeout(m_pHidDevice, data, 255, -1);
-
-        // This relieves that at the cost of higher CPU usage since we only
-        // block for a short while (500ms)
-        int result = hid_read_timeout(m_pHidDevice, data, 255, 500);
-        Trace timeout("HidReader timeout");
-        if (result > 0) {
-            Trace process("HidReader process packet");
-            //qDebug() << "Read" << result << "bytes, pointer:" << data;
-            QByteArray outData(reinterpret_cast<char*>(data), result);
-            emit(incomingData(outData, mixxx::Time::elapsed()));
-        }
-    }
-    delete [] data;
-}
+#define HIDCONTROLLER_POLL_LIMIT 10
```

Comment:
Replace with a typed `constexpr` in an anonymous namespace

Reply:
The maximum number of read loops might depend on the actual device. Therefore I suggest to make this an immutable member that is initialized at construction. The default value should be 1.

Reply:
So it should be configurable in the controller XML?

Reply:
No, mapping authors should not be concerned with such low level details. First we need to determine if a loop is actually necessary. Does repeatedly calling hid_read actually give different results or does it just repeat the most recently read data?

Reply:
I'm going to try that in the morning and will report back. 👍

Reply:
Looks like - at least, with my controller - that even if I call `hid_read` only once every 250ms, I get the latest message sent. I'm not sure if this is the same for every controller though. Pretty sure that there are controllers that work very differently in terms of which packets are received and such.

Reply:
Considering this behavior is basically the same as what the `HidReader` thread did before, I think it's fair to say that the right approach is to have no loop?

Reply:
> even if I call hid_read only once every 250ms, I get the latest message sent

But if you call it multiple times without blocking, does each call return the same data?

Reply:
Just tested, no, the data isn't always the same.